Test result:
```
====================================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.11.4
session_id:       2025-11-17--001
run time:         345 minutes 0.396 seconds
tests run:        153
passed:           151
flaky:            0
failed:           2
ignored:          0
====================================================================================================
...
----------------------------------------------------------------------------------------------------
test_id:    kafkatest.tests.streams.streams_smoke_test.StreamsSmokeTest.test_streams.processing_guarantee=exactly_once_v2.crash=False.metadata_quorum=ZK
status:     FAIL
run time:   10 minutes 24.823 seconds


    TimeoutError("Never saw 'REBALANCING -> RUNNING' message ducker@ducker08")
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/ducktape/tests/runner_client.py", line 186, in _do_run
    data = self.run_test()
  File "/usr/local/lib/python3.8/dist-packages/ducktape/tests/runner_client.py", line 246, in run_test
    return self.test_context.function(self.test)
  File "/usr/local/lib/python3.8/dist-packages/ducktape/mark/_mark.py", line 433, in wrapper
    return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
  File "/opt/kafka-dev/tests/kafkatest/tests/streams/streams_smoke_test.py", line 95, in test_streams
    monitor3.wait_until('REBALANCING -> RUNNING',
  File "/usr/local/lib/python3.8/dist-packages/ducktape/cluster/remoteaccount.py", line 753, in wait_until
    return wait_until(lambda: self.acct.ssh("tail -c +%d %s | grep '%s'" % (self.offset + 1, self.log, pattern),
  File "/usr/local/lib/python3.8/dist-packages/ducktape/utils/util.py", line 58, in wait_until
    raise TimeoutError(err_msg() if callable(err_msg) else err_msg) from last_exception
ducktape.errors.TimeoutError: Never saw 'REBALANCING -> RUNNING' message ducker@ducker08

----------------------------------------------------------------------------------------------------
...
----------------------------------------------------------------------------------------------------
test_id:    kafkatest.tests.streams.streams_broker_compatibility_test.StreamsBrokerCompatibility.test_fail_fast_on_incompatible_brokers_if_eos_v2_enabled.broker_version=1.1.1
status:     FAIL
run time:   2 minutes 6.181 seconds


    TimeoutError('Never saw \'Shutting down because the Kafka cluster seems to be on a too old version. Setting `processing.guarantee="exactly_once_v2"/"exactly_once_beta"` requires broker version 2.5 or higher.\' log message ducker@ducker04')
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/ducktape/tests/runner_client.py", line 186, in _do_run
    data = self.run_test()
  File "/usr/local/lib/python3.8/dist-packages/ducktape/tests/runner_client.py", line 246, in run_test
    return self.test_context.function(self.test)
  File "/usr/local/lib/python3.8/dist-packages/ducktape/mark/_mark.py", line 433, in wrapper
    return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
  File "/opt/kafka-dev/tests/kafkatest/tests/streams/streams_broker_compatibility_test.py", line 170, in test_fail_fast_on_incompatible_brokers_if_eos_v2_enabled
    log.wait_until('Shutting down because the Kafka cluster seems to be on a too old version. Setting processing\.guarantee="exactly_once_v2"/"exactly_once_beta" requires broker version 2\.5 or higher\.',
  File "/usr/local/lib/python3.8/dist-packages/ducktape/cluster/remoteaccount.py", line 753, in wait_until
    return wait_until(lambda: self.acct.ssh("tail -c +%d %s | grep '%s'" % (self.offset + 1, self.log, pattern),
  File "/usr/local/lib/python3.8/dist-packages/ducktape/utils/util.py", line 58, in wait_until
    raise TimeoutError(err_msg() if callable(err_msg) else err_msg) from last_exception
ducktape.errors.TimeoutError: Never saw 'Shutting down because the Kafka cluster seems to be on a too old version. Setting `processing.guarantee="exactly_once_v2"/"exactly_once_beta"` requires broker version 2.5 or higher.' log message ducker@ducker04

----------------------------------------------------------------------------------------------------

```

--- a/tests/docker/Dockerfile
+++ b/tests/docker/Dockerfile
@@ -62,8 +62,9 @@ LABEL ducker.creator=$ducker_creator
 # Update Linux and install necessary utilities.
 RUN apt update && apt install -y sudo git netcat iptables rsync unzip wget curl jq coreutils openssh-server net-tools vim python3-pip python3-dev libffi-dev libssl-dev cmake pkg-config libfuse-dev iperf traceroute iproute2 iputils-ping && apt-get -y clean
 RUN python3 -m pip install -U pip==21.1.1;
-# NOTE: ducktape 0.11.4 requires python3.9
-RUN pip3 install --upgrade cffi virtualenv pyasn1 boto3 pycrypto pywinrm ipaddress enum34 debugpy && pip3 install --upgrade "ducktape==0.12"
+# NOTE: ducktape 0.12.0 supports py 3.9, 3.10, 3.11 and 3.12
+COPY requirements.txt requirements.txt
+RUN pip3 install --upgrade -r requirements.txt
 
 COPY --from=build-native-image /build/kafka-binary/ /opt/kafka-binary/
 # Set up ssh

--- a/tests/docker/Dockerfile
+++ b/tests/docker/Dockerfile
@@ -147,7 +147,7 @@ RUN useradd -u $UID -ms /bin/bash ducker \
   && mkdir -p /home/ducker/ \
   && rsync -aiq /root/.ssh/ /home/ducker/.ssh \
   && chown -R ducker /home/ducker/ /mnt/ /var/log/ \
-  && echo "PATH=$(runuser -l ducker -c 'echo $PATH'):$JAVA_HOME/bin" >> /etc/environment \
+  && echo "PATH=$(runuser -l ducker -c 'echo $PATH'):$JAVA_HOME/bin" >> /home/ducker/.ssh/environment \
   && echo 'PATH=$PATH:'"$JAVA_HOME/bin" >> /home/ducker/.profile \
   && echo 'ducker ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers
 

--- a/tests/docker/Dockerfile
+++ b/tests/docker/Dockerfile
@@ -63,7 +63,7 @@ LABEL ducker.creator=$ducker_creator
 RUN apt update && apt install -y sudo git netcat iptables rsync unzip wget curl jq coreutils openssh-server net-tools vim python3-pip python3-dev libffi-dev libssl-dev cmake pkg-config libfuse-dev iperf traceroute iproute2 iputils-ping && apt-get -y clean
 RUN python3 -m pip install -U pip==21.1.1;
 # NOTE: ducktape 0.11.4 requires python3.9
-RUN pip3 install --upgrade cffi virtualenv pyasn1 boto3 pycrypto pywinrm ipaddress enum34 debugpy && pip3 install --upgrade "ducktape==0.11.4"
+RUN pip3 install --upgrade cffi virtualenv pyasn1 boto3 pycrypto pywinrm ipaddress enum34 debugpy && pip3 install --upgrade "ducktape==0.12"
 
 COPY --from=build-native-image /build/kafka-binary/ /opt/kafka-binary/
 # Set up ssh

--- a/tests/docker/ducker-ak
+++ b/tests/docker/ducker-ak
@@ -46,7 +46,7 @@ default_num_nodes=14
 
 # The default JDK base image with apt-get support.
 # The openjdk image has been officially deprecated. For more information, see: https://hub.docker.com/_/openjdk
-default_jdk="eclipse-temurin:8-jdk-focal"
+default_jdk="sapmachine:11-jdk-ubuntu-jammy"
 
 # The default ducker-ak image name.
 default_image_name="ducker-ak"

--- a/tests/docker/requirements.txt
+++ b/tests/docker/requirements.txt
@@ -1,0 +1,25 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cffi 
+virtualenv 
+pyasn1 
+boto3 
+pycrypto 
+pywinrm 
+ipaddress 
+debugpy 
+psutil
+ducktape==0.12.0

--- a/tests/setup.py
+++ b/tests/setup.py
@@ -51,7 +51,7 @@ setup(name="kafkatest",
       license="apache2.0",
       packages=find_packages(),
       include_package_data=True,
-      install_requires=["ducktape==0.11.4", "requests==2.31.0"],
+      install_requires=["ducktape==0.12", "requests==2.31.0", "psutil==5.7.2"],
       tests_require=["pytest", "mock"],
       cmdclass={'test': PyTest},
       zip_safe=False

